internal/repos/perforce.go

```
@@ -49,7 +49,32 @@ func newPerforceSource(svc *types.ExternalService, c *schema.PerforceConnection)
 // from the subsequent calls. This is going to be expanded as part of issue #44683
 // to actually only return true if the source can serve requests.
 func (s PerforceSource) CheckConnection(ctx context.Context) error {
-	return nil
+	// pick one depot to check for connectivity
+	for _, depot := range s.config.Depots {
+		// Tiny optimization: exit early if context has been canceled.
+		if err := ctx.Err(); err != nil {
+			return errors.Wrap(err, "context canceled")
+		}
+		u := url.URL{
+			Scheme: "perforce",
+			Host:   s.config.P4Port,
+			Path:   depot,
+			User:   url.UserPassword(s.config.P4User, s.config.P4Passwd),
+		}
+		p4Url, err := vcs.ParseURL(u.String())
+		if err != nil {
+			continue
```

Hmmmm, I think we should return the error here, no?

---

No, we want to try additional repos ins case the error is with the repo and not with the code host. But that approach is outdated now.